scenarios/two_party_countersigning/src/main.rs

```diff
@@ -252,6 +231,102 @@ fn agent_behaviour_initiate(
     Ok(())
 }
 
+#[allow(clippy::too_many_arguments)]
+async fn run_initiated_session(
+    client: TryCPClient,
+    my_preflight_response: PreflightResponse,
+    session_timeout: Instant,
+    agent_pub_key: AgentPubKey,
+    agent_name: String,
+    cell_id: CellId,
+    app_port: u16,
+    start: Instant,
+    initiated_success: Arc<AtomicUsize>,
+    reporter: Arc<Reporter>,
+) -> anyhow::Result<()> {
+    loop {
+        // Now listen for a signal from the remote with their acceptance
+        let signal = tokio::time::timeout_at(session_timeout, client.recv_signal()).await.with_context(|| format!("Agent [{agent_pub_key:?}] did not respond to the countersigning request in time, abandoning"))?;
+
+        match signal {
+            Some(signal) => {
+                let signal = match rmp_serde::decode::from_slice::<Signal>(&signal.data).map_err(
+                    |e| anyhow::anyhow!("Decoding failure, appears to not be a signal: {:?}", e),
+                )? {
+                    Signal::App { signal, .. } => signal,
+                    _ => {
+                        log::debug!("Received a signal that is not an app signal, listening for other signals.");
+                        continue;
+                    }
+                };
+
+                let other_response = match signal.clone().into_inner().decode::<Signals>() {
+                    Ok(Signals::Response(response))
+                        if session_fingerprint(response.request())?
```

Comment:
This changed. Previously I was comparing the `app_entry_hash` which isn't unique and was causing crossed wires and so validation failures. This is how Holochain checks that sessions are equal and massively improves the scenario behavior.